src/app/components/todo/TodoItems.tsx

```diff
@@ -0,0 +1,69 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { MouseEvent, useCallback } from "react";
+import TodoItem from "./TodoItem";
+import { MutationContext, Todo } from "../../types/todo";
+import { patchItem } from "../../api/todos";
+
+interface InputProps {
+  dataList: Todo[];
+}
+
+export default function TodoItems({ dataList }: InputProps) {
+  const queryClient = useQueryClient();
+
+  const toggleMutation = useMutation<
+    void,
+    Error,
+    Pick<Todo, "id" | "isCompleted">,
+    MutationContext
+  >({
+    mutationFn: ({ id, isCompleted }) => patchItem(id, { isCompleted }),
```

Comment:
여기 id, isCompleted가 any 타입으로 추론되고있는데 타입 안정성을 위해 함수 인자 타입을 올바르게 설정해주시면 좋을것같네요! 

```suggestion
    mutationFn: ({ id, isCompleted }: Pick<Todo, "id" | "isCompleted">) => patchItem(id, { isCompleted }),
```